### ISSUE TITLE
Consistent HTTPS detection

### ISSUE THREAD

**Issue by Korkman** [revive-adserver/revive-adserver#1057]
In several places only $_SERVER['HTTPS'] is used to detect https protocol. For example this breaks plugin installation in upgrade wizard when served via an HTTPS reverse proxy that only sets X-Forwarded-Proto or similar headers.

In other places, those headers are properly checked, giving the impression everything was fine until the upgrade wizard is used.

> Installation of plugin "openXReports" failed:
> Server returned: 0

Examining the AJAX calls shows mixed-content is blocked, therefore it fails.

I suggest simplifying the fix by setting $_SERVER['HTTPS'] = 'on' at a central location, then base all checks on that. This also fixes third party libraries like pear/Pager, pear/Net.

**Comment by mbeccati** [comment]
I think it's up to the webserver to properly send the HTTPS flag.

**Comment by Korkman** [comment]
> I think it's up to the webserver to properly send the HTTPS flag.

While it can be configured to do so, it rarely is. Therefore the header checks at various locations within Revive. At least it should be consistent so it doesn't give different results in different tasks.

**Comment by mbeccati** [comment]
That would also bring a matter of trusting headers or not. My preference would be to leave that outside of the application.

**Comment by Korkman** [comment]
True. Security wise an attacker could fool Revive to stick with HTTP although it is configured to redirect to HTTPS. The thing is, currently those headers are mostly trusted in the code. Removing that behavior would break many setups. Making it conistent with a central configuration option with default on for existing setups and default off for new installations would be best IMHO.

**Comment by wikt0r** [comment]
Thank you @Korkman for opening this issue. 

I had to move Revive 5 to a new server because the current hosting server does not support PHP 7. 

I had to proxy all `/revive` path but I could now figure how to force Revive to serve with `https`. The proxy between our servers is plain `http` so Revive served with `http`.

When I find you issue, I had an idea.

I set `fastcgi_param HTTPS 'on';` in `nginx.conf` in the Revive handler location path and everything works. :)
